classes/Telemetry.php

```diff
@@ -0,0 +1,353 @@
+<?php
+/*******************************************************************************
+ * Copyright (c) 2019, Code Atlantic LLC
+ ******************************************************************************/
+
+if ( ! defined( 'ABSPATH' ) ) {
+	// Exit if accessed directly
+	exit;
+}
+
+/**
+ * Our telemetry class.
+ *
+ * Handles sending usage data back to our servers for those who have opted into our telemetry.
+ *
+ * @since 1.11.0
+ */
+class PUM_Telemetry {
+
+	public static function init() {
+		add_action( 'pum_daily_scheduled_events', array( __CLASS__, 'track_check' ) );
+		if ( is_admin() && current_user_can( 'manage_options' ) ) {
+			add_filter( 'pum_alert_list', array( __CLASS__, 'optin_alert' ) );
+			add_action( 'init', array( __CLASS__, 'optin_alert_check' ) );
+		}
+	}
+
+	/**
+	 * Prepares and sends data, if it is time to do so
+	 * @since 1.11.0
+	 */
+	public static function track_check() {
+		if ( self::is_time_to_send() ) {
+			$data = self::setup_data();
+			self::send_data( $data );
+			set_transient( 'pum_tracking_last_send', true, 6 * DAY_IN_SECONDS );
+		}
+	}
+
+	/**
+	 * Prepares telemetry data to be sent
+	 * @return array
+	 * @since 1.11.0
+	 */
+	public static function setup_data() {
+		global $wpdb;
+
+		// Retrieve current theme info
+		if ( get_bloginfo( 'version' ) < '3.4' ) {
+			$theme_data = get_theme_data( get_stylesheet_directory() . '/style.css' );
+			$theme      = $theme_data['Name'] . ' ' . $theme_data['Version'];
+		} else {
+			$theme_data = wp_get_theme();
+			$theme      = $theme_data->Name . ' ' . $theme_data->Version;
+		}
+
+		// Retrieve current plugin information
+		if ( ! function_exists( 'get_plugins' ) ) {
+			include ABSPATH . '/wp-admin/includes/plugin.php';
+		}
+
+		$plugins        = array_keys( get_plugins() );
+		$active_plugins = get_option( 'active_plugins', array() );
+
+		foreach ( $plugins as $key => $plugin ) {
+			if ( in_array( $plugin, $active_plugins ) ) {
+				// Remove active plugins from list so we can show active and inactive separately
+				unset( $plugins[ $key ] );
+			}
+		}
+
+		$popups = 0;
+		foreach ( wp_count_posts( 'popup' ) as $status ) {
+			$popups += $status;
+		}
+
+		$popup_themes = 0;
+		foreach ( wp_count_posts( 'popup_theme' ) as $status ) {
+			$popup_themes += $status;
+		}
+
+		// Aggregates important settings across all popups.
+		$all_popups = pum_get_all_popups();
+		$triggers   = array();
+		$cookies    = array();
+		$conditions = array();
+		$location   = array();
+		$sizes      = array();
+		$sounds     = array();
+
+		// Cycle through each popup
+		foreach ( $all_popups as $popup ) {
+			$settings = $popup->get_settings();
+
+			// Cycle through each trigger to count the number of unique triggers.
+			foreach ( $settings['triggers'] as $trigger ) {
+				if ( isset( $triggers[ $trigger['type'] ] ) ) {
```

Comment:
Just out of curiousity, was this structured to follow a guideline, or could these all be set with a tertiary operand?

```
$triggers[ $trigger['type'] ] = isset( $triggers[ $trigger['type'] ] ) ? $triggers[ $trigger['type'] ] + 1 : 1
```

Reply:
To be honest, I just always forget about ternary operators. When I first started with WP, they were grouped in the coding standards as the "clever code" section where it states to avoid so I just never used them originally so now I don't think about them. But yeah, that could work here too.